### DISCUSSION TITLE
What are the options for note taking?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#1196] [Ideas] [answered]
Are there any features for note taking similar to org mode?

**Comment by NNBnh** [comment, marked as answer]
From `vision.md`:
> **Don't try to be everything for everyone:** There are many great editors out there to choose from. Let's make Helix one of those great options, with its own take on things.

Many people (include me) found open up and writing on a markup language file like Markdown, AsciiDoc or [ORML](https://github.com/NNBnh/orml) work just fine. So I afraid that this type of features won't be a built-in features.

Or maybe i'm don't know what i'm talking about, Org-mode is a document format right? Could you list some note taking features that you like to see on Helix?

**Reply by txtyash** [reply]
Ummm.... I can't just simply tell. It could a lot of different things to everyone.
[Watch this short demo video](https://www.youtube.com/watch?v=hnMntOQjs7Q)

[And this one is more detailed](https://www.youtube.com/watch?v=Ea_-TaEGa7Y)

Overall I'd say that note taking, todos and diary keeping are probably the well suited things for any text editor but all they tend to have is simple text editing. Org looks very intuitive. It is very close to taking organised notes with pen and paper. I'd love to see something similar.

**Reply by txtyash** [reply]
BTW, It is funny looking at how fast, responsive, and clean helix is that we might even be able to have something magnitudes better than org mode.

**Reply by txtyash** [reply]
So is it going to be implemented or any plans? @archseer

**Reply by archseer** [reply]
So... [orgmode](https://github.com/bzg/org-mode/tree/main/lisp) is as big as all of helix put together. (See also the [nvim port](https://github.com/nvim-orgmode/orgmode/tree/master/lua/orgmode).) I simply don't have time to work on something that big since there's lots of other helix work to be done. (I also personally just use Markdown to write down notes.)

@zim0369 if you're interested in implementing this as a plugin or a sub-crate I recommend you start with an RFC so we can figure out the full scope of work, then we can expose any missing APIs the feature might need.

**Reply by pickfire** [reply]
I do note taking with my own TODO format. Like
```
x done
o working on
- left
```

At least I don't see that I need org mode.

**Comment by slinlee** [comment]
While all of org-mode would be pretty fancy, these small improvements would help my note taking experience in helix using markdown:

- soft-wrapping for text
- automatically adding the right style of bullet (or todo item)
- shortcut for toggling todo.

As it is, I've already switched over to basic note taking in helix even though I miss some things from vimwiki.

**Reply by edrex** [reply]
> I've already switched over to basic note taking in helix even though I miss some things from vimwiki.

Same. 

> soft-wrapping for text

Implemented in #2184  `editor.soft-wrap.enable = true;` (also works per-language, see #6680)

> automatically adding the right style of bullet (or todo item)
> shortcut for toggling todo.

These could be implemented as LSP code actions, maybe with language-specific bindings to make them more ergonomic (see #5737).

**Reply by adriangalilea** [reply]
shortcut for toggling todo is why I arrived here

**Comment by edrex** [comment]
- Bringing @ubitux over from #2825. 
- Related: [Helix for prose #5830](https://github.com/helix-editor/helix/discussions/5830)

**Comment by getreu** [comment]
[Jens Getreu's blog - Note talking with Helix, Tp-Note and LanguageTool](https://blog.getreu.net/20220828-tp-note-new8/)

**Comment by luccahuguet** [comment]
an interesting development: https://github.com/Feel-ix-343/markdown-oxide

> Implementing Obsidian PKM features (and possibly more) in the form of a language server allows us to use these features in our favorite text editors and reuse other LSP-related plugins (like Telescope, outline, refactoring tools, ...). This language server primarily attempts to be a replacement and extension to the Obsidian markdown editor for Obsidian vault PKM-ing in your favorite text editing system -- creating the best PKM system for any text editor.
